### PR TITLE
Fixing string formatting in add package command

### PR DESCRIPTION
**Pull request by mishra14**
## Bug
Link: Internal email
Regression: No  

## Fix
Details: In an internal email, I saw that some exception messages in dotnet add command were not correctly formatted. This PR adds culture and right params to those exceptions.

## Testing/Validation
Tests Added: No  
Reason for not adding tests:  Changes to string formatting
Validation done:  Local testing with dotnet.exe

### CODE REVIEW

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 1]
```diff
@@ -475,7 +475,7 @@ For more information, visit http://docs.nuget.org/docs/reference/command-line-re
 {1} - Project path</comment>
   </data>
   <data name="Error_NoProjectFound" xml:space="preserve">
-    <value>Error while adding package '{0}' to project '{1}'. dotnet add package command supports adding package to one project at a time. But no project was found at the project path '{0}'.</value>
+    <value>Error while adding package '{0}' to project '{1}'. dotnet add package command supports adding package to one project at a time. But no project was found at the project path '{1}'.</value>
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: adding **a** package

or just remove the adding a package part, this should be obvious.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 2]
```diff
@@ -57,13 +59,25 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
             // Throw otherwise since we cannot proceed further.
             if (!matchingPackageSpecs.Any())
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: Use .Length == 0

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 3]
```diff
@@ -57,13 +59,25 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
...
             }
             else if (matchingPackageSpecs.Count() > 1)
             {
-                packageReferenceArgs.Logger.LogDebug(Strings.Error_MultipleMatchingSpecs);
-                throw new Exception(Strings.Error_MultipleProjectsFound);
+                packageReferenceArgs.Logger.LogDebug(string.Format(Strings.Error_MultipleMatchingSpecs,
+                    packageReferenceArgs.PackageDependency.Id,
+                    packageReferenceArgs.ProjectPath));
+
+                throw new Exception(string.Format(CultureInfo.CurrentCulture,
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: Use `InvalidOperationException` instead or make a new type specific to dotnet add.

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 4]
```diff
@@ -41,7 +41,9 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
             var dgSpec = ReadProjectDependencyGraph(packageReferenceArgs);
             if (dgSpec == null)
             {
-                throw new Exception(Strings.Error_NoDgSpec);
+                packageReferenceArgs.Logger.LogDebug(Strings.Error_NoDgSpec);
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: Add a comment here on loc

**Review by emgarten — CHANGES_REQUESTED** [review]

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 5]
```diff
@@ -458,24 +458,13 @@ For more information, visit http://docs.nuget.org/docs/reference/command-line-re
...
   </data>
-  <data name="Error_MultipleProjectsFound" xml:space="preserve">
-    <value>Error while adding package '{0}' to project '{1}'. dotnet add package command supports adding package to one project at a time. But multiple projects were found at the project path '{1}'.</value>
-    <comment>{0} - Package Id
-{1} - Project path</comment>
-  </data>
-  <data name="Error_NoProjectFound" xml:space="preserve">
-    <value>Error while adding package '{0}' to project '{1}'. dotnet add package command supports adding package to one project at a time. But no project was found at the project path '{0}'.</value>
+  <data name="Error_UnsupportedProject" xml:space="preserve">
+    <value>Error while adding package '{0}' to project '{1}'. The project does not support adding package reference through add package command.</value>
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: reference*s*

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 6]
```diff
@@ -55,15 +58,12 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
 
             // This ensures that the DG specs generated in previous steps contain exactly 1 project with the same path as the project requesting add package.
             // Throw otherwise since we cannot proceed further.
-            if (!matchingPackageSpecs.Any())
-            {
-                packageReferenceArgs.Logger.LogDebug(Strings.Error_NoMatchingSpecs);
-                throw new Exception(Strings.Error_NoProjectFound);
-            }
-            else if (matchingPackageSpecs.Count() > 1)
+            if (matchingPackageSpecs.Length != 1)
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs:61]: 💯

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs [hunk 7]
```diff
@@ -41,7 +41,10 @@ public class AddPackageReferenceCommandRunner : IPackageReferenceCommandRunner
             var dgSpec = ReadProjectDependencyGraph(packageReferenceArgs);
             if (dgSpec == null)
             {
-                throw new Exception(Strings.Error_NoDgSpec);
+                // Logging non localized error on debug stream.
+                packageReferenceArgs.Logger.LogDebug(Strings.Error_NoDgSpec);
+
+                throw new Exception(string.Format(CultureInfo.CurrentCulture, Strings.Error_NoDgSpec));
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Commands/PackageReferenceCommands/AddPackageReferenceCommandRunner.cs]: InvalidOperationException here also

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs [hunk 8]
```diff
@@ -60,7 +56,7 @@ private static Project GetProject(string projectCSProjPath, IDictionary<string,
             var projectRootElement = TryOpenProjectRootElement(projectCSProjPath);
             if (projectCSProjPath == null)
             {
-                throw new Exception(Strings.Error_MsBuildUnableToOpenProject);
+                throw new Exception(string.Format(CultureInfo.CurrentCulture, Strings.Error_MsBuildUnableToOpenProject, projectCSProjPath));
```
Comment (emgarten) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Utility/MSBuildAPIUtility.cs]: I would also make these InvalidOperationExceptions, it is only slightly more useful, but it at least could be caught without catching ALL exceptions.

**Review by emgarten — APPROVED** [review]
Address feedback then :shipit:

File: src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx [hunk 9]
```diff
@@ -458,24 +458,13 @@ For more information, visit http://docs.nuget.org/docs/reference/command-line-re
...
   </data>
-  <data name="Error_MultipleProjectsFound" xml:space="preserve">
-    <value>Error while adding package '{0}' to project '{1}'. dotnet add package command supports adding package to one project at a time. But multiple projects were found at the project path '{1}'.</value>
-    <comment>{0} - Package Id
-{1} - Project path</comment>
-  </data>
-  <data name="Error_NoProjectFound" xml:space="preserve">
-    <value>Error while adding package '{0}' to project '{1}'. dotnet add package command supports adding package to one project at a time. But no project was found at the project path '{0}'.</value>
+  <data name="Error_UnsupportedProject" xml:space="preserve">
+    <value>Error while adding package '{0}' to project '{1}'. The project does not support adding package references through add package command.</value>
```
Comment (nkolev92) [inline src/NuGet.Core/NuGet.CommandLine.XPlat/Strings.resx]: through "the" add package command?

**Review by nkolev92 — APPROVED** [review]
Minor feedback.

**Pull request merged**